sys/trickle/trickle.c

```diff
@@ -33,25 +35,21 @@ void trickle_callback(trickle_t *trickle)
 
 void trickle_interval(trickle_t *trickle)
 {
-    uint32_t old_interval = trickle->I;
-    uint32_t max_interval = trickle->Imin << trickle->Imax;
-    uint32_t diff = old_interval - trickle->t;
+    assert(trickle->I > 0);
 
-    trickle->I *= 2;
+    uint32_t diff = trickle->I - trickle->t;
 
-    if ((trickle->I == 0) || (trickle->I > max_interval)) {
-        trickle->I = max_interval;
-    }
+    trickle->I *= 2;
+    trickle->I = MIN(trickle->I, (trickle->Imin << trickle->Imax));
```

Comment:
Still, we have a possibility for a corner case which must be handled.
No matter if the current setup prevents it.